lib/adapters/xhr.js

```diff
@@ -168,6 +168,21 @@ module.exports = function xhrAdapter(config) {
       request.upload.addEventListener('progress', config.onUploadProgress);
     }
 
+    // Check maxContentLength if needed
+    if (config.maxContentLength > 0) {
```

Comment:
In http.js line 220 the condition is `config.maxContentLength > -1`. Perhaps we should allow limit of 0 bytes here also.